File: src/atcoder/abc314/swift/abc314_d.swift
func get1() -> Int {
  return Int(readLine()!)!
}

struct Query {
  let t:Int
  let x:Int
  let c:Character
}

let n = get1()
var s = Array(readLine()!)
let q = get1()
let qs = (0..<q).map { _ -> Query in
  let line = readLine()!.split(separator: " ")
  return Query(
    t: Int(line[0])!,
    x: Int(line[1])! - 1,
    c: Array(line[2])[0]
  )
}

var used = false
let filtered_qs = qs.reversed().filter {
  if $0.t == 1 { return true }
  if used { return false }
  used = true
  return true
}

for query in filtered_qs.reversed() {
  if query.t == 1 {
    s[query.x] = query.c
  } else if query.t == 2 {
    s = Array(String(s).lowercased())
  } else {
    s = Array(String(s).uppercased())
  }
}
print(String(s))

File: src/atcoder/abc314/scala/abc314_d.scala
import io.StdIn.readLine
import scala.collection.mutable.ArrayBuffer

val buf = java.util.StringTokenizer(io.Source.stdin.mkString)
def sc = buf.nextToken

case class Query(t:Int, x:Int, c:Char)

object Main extends App:
  val n = sc.toInt
  var s = StringBuilder(sc)
  val q = sc.toInt
  val queries = ArrayBuffer[Query]()
  for _ <- 0 until q do
    val t = sc.toInt
    val x = sc.toInt - 1
    val c = sc(0)
    queries += Query(t, x, c)

  var used = false
  val filtered = queries.reverse.filter(q => {
    if q.t == 1 then
      true
    else if used then
      false
    else
      used = true
      true
  }).reverse

  for query <- filtered do
    query.t match {
      case 1 => s(query.x) = query.c
      case 2 => s = StringBuilder(s.toString.toLowerCase)
      case 3 => s = StringBuilder(s.toString.toUpperCase)
    }

  println(s.toString)
